### PR TITLE
propagation chart

### LINKED ISSUE #86: propagation chart
Chart can plot the time diff values for the votes.

### PR DESCRIPTION
**Pull request by ademuanthony**
Closes #86 

image

image

### CODE REVIEW

File: web/server.go [hunk 1]
```diff
@@ -109,7 +111,7 @@ func (s *Server) registerHandlers(r *chi.Mux) {
 	r.Get("/", s.homePage)
 	r.Get("/exchanges", s.getExchangeTicks)
 	r.Get("/filteredEx", s.getFilteredExchangeTicks)
-	r.Get("/chartExchange", s.getChartData)
+	r.Get("/chartExchange", s.getExchangeChartData)
```
Comment (Sirmorrison) [inline web/server.go]: Revert this change if it doesnt affect this issue as it will result int a major conflict in #117

**Comment by Sirmorrison** [comment]
image

Please fix the issue with the circled area when the screen is reduced.

File: web/handlers.go [hunk 2]
```diff
@@ -667,6 +668,53 @@ func (s *Server) getPropagationData(res http.ResponseWriter, req *http.Request)
...
+	for _, record := range data {
+		if existingTime, found := avgTimeForHeight[record.BlockHeight]; found {
+			avgTimeForHeight[record.BlockHeight] = (record.TimeDifference + existingTime)/2
+		} else {
+			avgTimeForHeight[record.BlockHeight] = record.TimeDifference
+			heightArr = append(heightArr, record.BlockHeight)
+		}
+	}
+
+	var yLabel = "Delay (s)"
```
[the remark this answers is not in the document]
Comment (ademuanthony) [reply web/handlers.go]: english expects space before `(`

File: web/handlers.go [hunk 3]
```diff
@@ -667,6 +668,53 @@ func (s *Server) getPropagationData(res http.ResponseWriter, req *http.Request)
...
+	if requestedRecordSet == "votes" {
+		yLabel = "Time Difference (s)"
+	}
+	var csv = fmt.Sprintf("Height,%s\n", yLabel)
```
[the remark this answers is not in the document]
Comment (ademuanthony) [reply web/handlers.go]: This is `CSV` and shouldn't come with any unwanted whitespace
https://discuss.codecademy.com/t/in-a-csv-file-does-spacing-matter-before-and-after-commas/353056
